### ISSUE TITLE
small edits in build.sh may be needed

### ISSUE THREAD

**Issue by ngam** [conda-forge/tensorflow-feedstock#250]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

In build.sh, `export CC_OPT_FLAGS="-O2"` is overridden by `export CC_OPT_FLAGS="-march=nocona -mtune=haswell"`, if the `-O2` it should be carried through.

Why can't we do `-Ofast` here btw?

### Installed packages

```shell
n/a
```

### Environment info

```shell
m/a
```

**Comment by ngam** [comment]
Note: we cannot pass `-Ofast`, it fails XLA error like:

https://github.com/tensorflow/tensorflow/issues/27942

**Comment by xhochy** [comment]
We shouldn't use this environment variable at all as it interferes with the bazel toolchain. O2 is just a generic placeholder as it cannot be empty.

**Comment by ngam** [comment]
> We shouldn't use this environment variable at all as it interferes with the bazel toolchain. O2 is just a generic placeholder as it cannot be empty.

I used it locally and it was okay, except for `-Ofast`. It was also printed in the calls.

I also used the other one, `CC_OPT_FLAGS="-march=nocona -mtune=haswell"`, and when set to avx512, it works (i.e. is segfaults on non-avx512 machines).

But I am not really sure if these two edits are worth it. I am just highlighting it here in case someone thought either flag is important for something.

**Comment by xhochy** [comment]
You should be using standard CFLAGS instead. If you don't cross-compile it probably doesn't make a difference.

**Comment by ngam** [comment]
What do you mean? What's the point of this then? https://github.com/conda-forge/tensorflow-feedstock/blob/139cd3242341f27b314f3f997ad46429c912fa69/recipe/build.sh#L138

**Comment by ngam** [comment]
Are you saying that `CC_OPT_FLAGS` doesn't make a difference if not cross-compiling? Or are you saying `CC_OPT_FLAGS=-Ofast` is the same as passing `-Ofast` (as a standard CFLAG)?

Idk, but it seems like it does make a difference --- I opened and closed an issue in bazel-toolchain about this. I realized the failures were not to do with bazel (hence closed) but instead, `-march=` cannot be higher than the native build machine. I was trying to super-tune the tensorflow build (btw to really minimal performance advantage sadly... at least in my workflow).

https://github.com/conda-forge/bazel-toolchain-feedstock/issues/11

(I will follow up on the ppc issue separately later)

**Comment by xhochy** [comment]
> What do you mean? What's the point of this then?
> 
> https://github.com/conda-forge/tensorflow-feedstock/blob/139cd3242341f27b314f3f997ad46429c912fa69/recipe/build.sh#L138

This should be deleted.

**Comment by xhochy** [comment]
> Are you saying that `CC_OPT_FLAGS` doesn't make a difference if not cross-compiling? Or are you saying `CC_OPT_FLAGS=-Ofast` is the same as passing `-Ofast` (as a standard CFLAG)?

`CC_OPT_FLAGS` is used by the build but by builds for host and target. Thus if you cross-compile and add any architecture-specific options, it would break the build. Setting `CFLAGS` correctly before invoking `source gen-bazel-toolchain` ensures that `CFLAGS` are only used for the target systems.

**Comment by ngam** [comment]
Makes sense. It currently works only because it is set for cuda (no cross-compile). But we should set CFLAGS correctly going forward then. Thanks!
